compiler/rustc_middle/src/traits/chalk.rs

```diff
@@ -336,27 +333,16 @@ pub enum ChalkEnvironmentClause<'tcx> {
     TypeFromEnv(Ty<'tcx>),
 }
 
-impl<'tcx> TypeFoldable<'tcx> for &'tcx ty::List<ChalkEnvironmentClause<'tcx>> {
-    fn super_fold_with<F: TypeFolder<'tcx>>(&self, folder: &mut F) -> Self {
-        let v = self.iter().map(|t| t.fold_with(folder)).collect::<SmallVec<[_; 8]>>();
```

Comment:
This line is the same in method used instead (existing implementation of trait on `&'tcx ty::List<Predicate<'tcx>>`, it's just that a different method for `intern_chalk_environment_clause_list` gets called.